examples/splinterdb_custom_ipv4_addr_sortcmp_example.c

```diff
@@ -0,0 +1,456 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * ----------------------------------------------------------------------------
+ * SplinterDB Advanced Iterators Example Program with custom sort-comparison.
+ *
+ * What's new beyond previous splinterdb_iterators_example.c?
```

Comment:
I think this program makes the assumption that people will consume our examples in some sort of linear fashion. 
I found this assumption very weak (for example, I rarely do that).